grid2op/Chronics/GSFFWFWM.py

```diff
@@ -0,0 +1,148 @@
+# Copyright (c) 2019-2020, RTE (https://www.rte-france.com)
+# See AUTHORS.txt
+# This Source Code Form is subject to the terms of the Mozilla Public License, version 2.0.
+# If a copy of the Mozilla Public License, version 2.0 was not distributed with this file,
+# you can obtain one at http://mozilla.org/MPL/2.0/.
+# SPDX-License-Identifier: MPL-2.0
+# This file is part of Grid2Op, Grid2Op a testbed platform to model sequential decision making in power systems.
+import os
+import json
+import numpy as np
+import pandas as pd
+from datetime import datetime, timedelta
+
+
+from grid2op.dtypes import dt_bool, dt_int
+from grid2op.Chronics.GridStateFromFileWithForecasts import GridStateFromFileWithForecasts
+import pdb
+
+
+class GridStateFromFileWithForecastsWithMaintenance(GridStateFromFileWithForecasts):
+    """
+    An extension of :class:`GridStateFromFileWithForecasts` that implements the maintenance chronic generator
+    on the fly (maintenance are not read from files, but are rather generated when the chronics is created).
+
+    Attributes
+    ----------
+    maintenance_starting_hour: ``int``
+        The hour at which every maintenance will start
+
+    maintenance_ending_hour: ``int``
+        The hour at which every maintenance will end (we suppose mainteance end on same day for now
+
+    line_to_maintenance: ``array``, dtype: ``string``
+        Array used to store the name of the lines that can happen to be in maintenance
+
+    daily_proba_per_month_maintenance: ``array``, dtype: ``float``
+        Array used to store probability each line can be in maintenance on a day for a given month
+
+    max_daily_number_per_month_maintenance: ``array``, dtype: ``int``
+        Array used to store maximum number of maintenance per day for each month
+
+    """
+
+    def initialize(self, order_backend_loads, order_backend_prods, order_backend_lines, order_backend_subs,
+                   names_chronics_to_backend=None):
+
+        self.name_line = order_backend_lines
+
+        # properties of maintenance
+        # self.maintenance_duration= 8*(self.time_interval.total_seconds()*60*60)#8h, 9am to 5pm
+        # 8h furation, 9am to 5pm
+        with open(os.path.join(self.path, "maintenance_meta.json"), "r", encoding="utf-8") as f:
```

Comment:
THis class should not be used with other datasets ;-) But i agree this could be done differently... Though handly corner cases is probably something we'll have to cover in 0.9.1 :-/